src/core/ddsi/src/ddsi_plist.c

```diff
@@ -1691,6 +1691,7 @@ static const struct piddesc piddesc_omg[] = {
   { PID_TYPE_CONSISTENCY_ENFORCEMENT, PDF_QOS | PDF_FUNCTION, QP_TYPE_CONSISTENCY_ENFORCEMENT, "TYPE_CONSISTENCY_ENFORCEMENT",
     offsetof (struct ddsi_plist, qos.type_consistency), membersize (struct ddsi_plist, qos.type_consistency),
     { .f = { .deser = deser_type_consistency, .ser = ser_type_consistency, .valid = valid_type_consistency, .equal = equal_type_consistency, .print = print_type_consistency } }, 0 },
+  QP  (DATA_REPRESENTATION,                 data_representation, XQ, XE2, XSTOP),
```

Comment:
Yes, an empty sequence is valid according to the spec, and should be considered as `XCDR` (version 1). Duplicates are not mentioned in the spec, but obviously it does not make sense to have duplicates in this list. But I don't think it is worth adding a custom validator function for that, as it is not really a problem if there are duplicates. 

While re-reading the spec, I realized that the range is not an enum and not limited to 0..2, but can be any int16 value, so `XE2` is incorrect and should be a int16 instruction (I used `Xs` to be consistent with other int types, but this may be confusing because of `XS` used for strings?)